### PR TITLE
br/lightning: unify SplitWaitScatter logic into `split` package

### LINKED ISSUE #51520: Need to check the region status before splitting regions
## Enhancement
Assume that we are importing a large table `t1`, the regions of `t1` are split from the region `a`. We'd better check the region `a`'s status to avoid that it is a lack of peers, even if the probability of this issue is small. If not, after importing the data, PD will spend much time making up the replicas which wastes resources and may cause data loss if one of TiKV crushes.

WuT4yrcoza
t3WGv3TKHd

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #51533 close #51520

Problem Summary:

### What changed and how does it work?

provide a `SplitWaitScatter` method for BR and lightning, see br/pkg/restore/split/client.go

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Leavrth — APPROVED** [review]

File: br/pkg/restore/split/client.go [hunk 1]
```diff
@@ -42,61 +43,52 @@ const (
...
+	// The split step has a few retry times. If it meets error, the error is returned
+	// directly.
+	//
+	// The split waiting step has a backoff retry logic, if split has made progress,
+	// it will not increase the retry counter. Otherwise, it will retry for about 1h.
+	// If the retry is timeout, it will log a warning and continue.
+	//
+	// The scatter step has a few retry times. If it meets error, it will log a
+	// warning and continue.
+	SplitWaitScatter(ctx context.Context, region *RegionInfo, keys [][]byte) (*RegionInfo, []*RegionInfo, error)
```
Comment (D3Hunter) [inline br/pkg/restore/split/client.go:67]: SplitAndWaitScatter?

**Review by D3Hunter — APPROVED** [review]

File: br/pkg/restore/split/client.go (hunk 1 above)
Comment (lance6716) [reply br/pkg/restore/split/client.go:67]: If we want to mention `And`, maybe `SplitAndWaitAndScatter` (I don't like the repeated `And`) or `SplitWaitAndScatter` (like in English "split, wait, and scatter"). Do you think `SplitWaitAndScatter` is better?

**Comment by lance6716** [comment]
/hold

wait for comments

File: br/pkg/restore/split/client.go (hunk 1 above)
Comment (D3Hunter) [reply br/pkg/restore/split/client.go:67]: lgtm

**Pull request merged**